providers/implementations/signature/ecdsa_sig.c

```diff
@@ -227,7 +227,7 @@ static int ecdsa_setup_md(PROV_ECDSA_CTX *ctx, const char *mdname,
     sha1_allowed = (ctx->operation != EVP_PKEY_OP_SIGN);
     md_nid = ossl_digest_get_approved_nid_with_sha1(ctx->libctx, md,
                                                     sha1_allowed);
-    if (md_nid == NID_undef) {
```

Comment:
Should this be `<=` rather than `<` ?

Reply:
No, because this is exactly the change that allows doing arbitrary digests with ECDSA as in the reported bug #14696